src/containers/GroupsContainer.js

```diff
@@ -132,6 +136,8 @@ const GroupsContainer = props => {
         updateUser={updateUser}
         tutorialWatched={user.tutorialWatched}
         teams={teams}
+        getPosts={getPosts}
+        posts={posts}
```

Comment:
`posts` are passed here as props, but it's never used in Groups-component

Reply:
I had some difficulty getting the posts to work here. Not sure why, but when I tried using them as props (these were left here after some such attempt) I couldn't reliably get them. Might have been just me not knowing how to do that properly :D

Reply:
I was tinkering around a little bit, and I think this produces the exact same number of unread messages as with getMessages function:

```
  useEffect(() => {
    if (unreadCount && unreadCount > 0 && propPosts) {
      const channelPosts =
        propPosts &&
        Object.values(propPosts).filter(item => item.channel_id === channel.id)
      const beginIndex = Object.keys(channelPosts).length - unreadCount
      const sorted = channelPosts
        .sort((p1, p2) => p1.create_at - p2.create_at)
        .slice(beginIndex)
      const filtered = sorted.filter(p => p.type === '')
      console.log(
        'Filtered props posts len',
        channel.display_name,
        filtered && filtered.length
      )
    }
  }, [propPosts, unreadCount])

```

Reply:
This is a bit silly, since current implementation works well, but I could not resist

Reply:
Tinker away! It's great if you come up with a better way of doing something. Feel free to push that :)

Reply:
So I did!

Reply:
Hmm, it doesn't.. If I join or leave the group, a message will disappear from the count. I had some trouble like this before, maybe that's the reason why I did it the way I did.